### DISCUSSION TITLE
ScriptReference/SerializedProperty.DeleteArrayElementAtIndex

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#98] [Notes]
# ScriptReference/SerializedProperty.DeleteArrayElementAtIndex

https://docs.unity3d.com/ScriptReference/SerializedProperty.DeleteArrayElementAtIndex.html

**Comment by vertxxyz** [comment]
When used on an [`objectReferenceValue`](https://docs.unity3d.com/ScriptReference/SerializedProperty-objectReferenceValue.html), if the element is not `null` this will set it to `null` and it won't be removed.
```csharp
if (property.GetArrayElementAtIndex(i).objectReferenceValue != null)
	property.DeleteArrayElementAtIndex(i); // Make reference null
property.DeleteArrayElementAtIndex(i); // Remove array element
```
